src/main/java/Freedom.java

```diff
@@ -0,0 +1,256 @@
+        final String START_MESSAGE = """
+                \tHello! I'm Freedom
+                \tWhat can I do for you?
+                """;
+        final String CLOSING_MESSAGE = "\tBye. Hope to see you again soon!\n";
+
+        System.out.println(LOGO + START_MESSAGE + LOGO);
+        Scanner in = new Scanner(System.in);
+
+        try {
+            checkData();
+            loadData();
+
+            // Super loop for getting inputs
+            while (in.hasNextLine()) {
+                String line = in.nextLine();
+
+                if (line.equals("bye")) {
+                    break;
+                }
+                handleInput(line);
+            }
+
+            saveData();
+        } catch (Exception e) {
+            System.out.print("");
+        }
+
+        System.out.println(LOGO + CLOSING_MESSAGE + LOGO);
+    }
+
+    public static void handleInput(String input) {
+        final int COMMAND_INDEX = 0;
+        String[] words = input.split(" ");
+        String description;
+
+        try {
+            switch (words[COMMAND_INDEX]) {
+                case "list":
+                    printList();
+                    return;
+                case "mark":
+                case "unmark":
+                case "delete":
+                    editTask(words);
+                    return;
+                case "todo":
+                    description = input.replaceFirst("todo", "");
+                    tasks.add(new ToDo(description));
+                    break;
+                case "deadline":
+                    description = input.replaceFirst("deadline", "");
+                    tasks.add(new Deadline(description));
+                    break;
+                case "event":
+                    description = input.replaceFirst("event", "");
+                    tasks.add(new Event(description));
+                    break;
+                default:
+                    throw new InvalidCommand();
+
+            }
+            System.out.println(LOGO + "\tGot it. I've added this task: ");
+            System.out.println("\t  " + tasks.get(lastIndex).printLine());
+            
+            lastIndex++;
+
+            System.out.println("\tNow you have " + (lastIndex) + " tasks in the list.\n" + LOGO);
+        } catch (InvalidCommand e) {
+            System.out.println(LOGO + "\tSorry! I do not understand your command");
+            System.out.println(LOGO);
+        } catch (Exception e) {
+            System.out.print("");
+        }
+    }
+
+    public static void printList() {
+        int counter = 1;
+        System.out.print(LOGO + "\tHere are the tasks in your list:\n");
+        for (int i = 0; i < lastIndex; i++) {
+            System.out.println("\t" + counter + "." + tasks.get(i).printLine());
+            counter++;
+        }
+        System.out.println(LOGO);
+    }
+
+    public static void editTask(String[] words) {
+        final int COMMAND_INDEX = 0;
```

Comment:
Avoid magic numbers. Instead of hardcoding indices like 0, 1, 2 in editTask, declare constants to improve readability.